docs/design/2024-01-22-ticdc-large-txn-not-block-wm.md

```diff
@@ -0,0 +1,205 @@
+
+![image](../media/large-txn-no-block-wm-4.png)
+
+### WatermarkAdvancer handles locks
+
+When there is a new added lock for a large transaction, this must be a prewrite request,
+
+- Write lock
+  - if the start-ts of the new added lock is already in current large_txn_map we can skip it directly since we already tracked this large transaction.
+  - If the start-ts of the new added lock is not in current large_txn_map , this must be the first prewrite key in this region(data range), we add it to the large_txn_map
+    When there is a new delete lock for a large transaction:
+- Delete lock, this means the transaction's final status is determined, either commited or rollbacked:
+  - If its start-ts is still in large_txn_map, this must be the first received committed/rollbacked key for this transaction in this data range(region), we can remove it from the large_txn_map
+  - If its start-ts is not in large_txn_map, this is not the first received committed/rollbacked key for this transaction in this data ragne(region), we can ignore it since the water mark might already advanced by the first delete-lock in this data range(region)
+
+### WatermarkAdvancer update min-commit-ts for large transactions
+
+If the large_txn_map is not empty, it means there must be at least one large transaction is ongoing in this data range(region), we need to periodically query the min-commit-ts for these ongoing large transactions to advance the watermark of this data range.
+
+![image](../media/large-txn-no-block-wm-5.png)
+
+The above diagram demonstrates a large transaction with start-ts = 100 write multiple data ranges. Each data range's WatermarkAdvancer will periodically query the min-commit-ts in its primary key for this large transaction.
+
+#### Performance Assessment (range-level request VS store-level request)
+
+You may have noticed that, letting each data range's WatermarkAdvancer proactively query the primary key of this large transaction may introduce a performance issue. For example, if the large transaction involves 100,000 or more data ranges, each WatermarkAdvancer queries the primary key once per second (in order to fetch fresh enough min-commit-ts), the primary key data range needs to serve 100,000 queries.
+In order to decline the query QPS by many magnititude, we use store level requests instead of data range level requests. We can reuse TiKV's 'store level check leader request' to achieve this goal. There are at most N requests per second for the primary key in the above example, where N is the number of TiKV nodes, let's say 200, which represents a relatively large cluster.
```

Comment:
```suggestion
In order to decline the query QPS by many magnitude, we use store level requests instead of data range level requests. We can reuse TiKV's 'store level check leader request' to achieve this goal. There are at most N requests per second for the primary key in the above example, where N is the number of TiKV nodes, let's say 200, which represents a relatively large cluster.
```